src/sensai/data_transformation/dft.py

```diff
@@ -491,110 +491,162 @@ class DFTNormalisation(DataFrameTransformer):
-                `transformer` is not given; if neither `transformer` nor `transformerInstance` are given, the containing instance's default
+                `transformer` is not given; if neither `transformer` nor `transformer_factory` are given, the containing ``DFTNormalisation`` instance's default
                 factory will be used. See :class:`SkLearnTransformerFactoryFactory` for convenient construction options.
-            :param independent_columns: whether, for the case where the rule matches multiple columns, the columns are independent and a
-                separate transformation is to be learned for each of them (rather than using the same transformation for all columns and
-                learning the transformation from the data of all columns); must be specified for rules matching more than one column,
-                None is acceptable only for a single column
+            :param array_valued: only allowed if the rule matches a single column. If True, it expresses that
+                column values are not scalars but arrays (of arbitrary lengths).
+                It is then assumed that all entries in such arrays are to be normalised in the same way, i.e., the same
+                transformation will be applied to each entry in the array.
+            :param fit: whether the rule's transformer shall be fitted. One use case for setting this to False is
+                if a transformer instance instead of a factory is given and the transformer is already fitted.
+            :param independent_columns: only relevant if the resulting rule matches multiple columns, in which case it is required.
+                If True, the columns are treated independent and a separate transformation is to be learned for each of them. Note that
+                this doesn't mean each column will get a separate transformer instance! Rather, the transformer will be fitted
+                on the array resulting from selecting the matched columns.
+                If False, all matching columns are treated as a single feature for the purpose of normalisation.
+                Thus, all columns will be concatenated before fitting the transformer.
             """
-            if (skip or unsupported) and count_not_none(transformer, transformer_factory) > 0:
-                raise ValueError("Passed transformer or transformerFactory while skip=True or unsupported=True")
-            self.skip = skip
-            self.unsupported = unsupported
-            self.transformer = transformer
-            self.transformerFactory = transformer_factory
-            self.independentColumns = independent_columns
+            super().__init__(skip=skip, unsupported=unsupported, transformer=transformer, transformer_factory=transformer_factory,
+                array_valued=array_valued, fit=fit, independent_columns=independent_columns)
 
-        def to_rule(self, regex: Optional[str]):
+        def to_rule(self, regex: Optional[str| re.Pattern[str]]):
```

Comment:
The `|` operator was introduced in Python 3.10; sensAI supports 3.7+, so it can't be used.
And `Pattern` didn't become generic until 3.9.

Reply:
Ah, man, I thought I'd removed them all. Good catch!